packages/utils/src/browser.ts

```diff
@@ -86,6 +86,11 @@ function _htmlElementAsString(el: unknown, keyAttrs?: string[]): string {
     return '';
   }
 
+  // If using the component name annotation plugin, this value may be available on the DOM node
+  if (elem instanceof HTMLElement && elem.dataset && elem.dataset['sentryComponent']) {
```

Comment:
l: We might have to check if `HTMLElement` is defined globally and if this will break typescript types when folks are running in Node only environments (utils is technically meant to be isomorphic).

I guess we can't extract this util out because replay relies on it cc @mydea

Reply:
As per feedback from Luca, can we check if dom types are included when we add this change? 

We can do that by looking at the type definitions of the built assets and checking for `dom.d.ts` being included.

Reply:
Added some checks to prevent breaking types when running in Node 👍